### DISCUSSION TITLE
Run M6 Wing in incompressible condition

### DISCUSSION THREAD

**Discussion by ayt1998** [mdolab/dafoam#279] [General]
Hello, thank you very much for the last question.
Firstly, the tutorial M6 Wing ran well with slover DArhoSimpleCFoam, the result were the same as on the website.
Then I tried to change the flow condition to incompressible and solveCL was successfully done.
I tried two optimizers ipopt and slsqp, but slsqp showed Cd increased rather than decreased and ipopt showed Cd didn't change.
![slsqp](https://user-images.githubusercontent.com/67814946/155490316-e424446a-054b-4b1d-854c-84987fe2d77e.png)
![ipopt](https://user-images.githubusercontent.com/67814946/155490348-f98e2ddb-356c-4596-a57a-2285ca9681af.png)

Can you come up with any solution?
Best regards.
@friedenhe

**Comment by friedenhe** [comment]
DArhoSimpleCFoam this solver works in transonic conditions only. If you want to do incompressible, use DASimpleFoam. Refer to the setup in NACA0012/incompressible

**Comment by ayt1998** [comment]
Yes, I used DASimpleFoam and forgot to mention it.

**Reply by friedenhe** [reply]
Check if the flow and adjoint are converging.

**Comment by ayt1998** [comment]
Thanks for your reminding. 
First, I executed runPrimal and solveCL successfully.
Then I ran opt, "Evaluating Objective Functions 002" in logOpt.txt showed some errors:
![2022-02-26 16-33-06屏幕截图](https://user-images.githubusercontent.com/67814946/155837009-0e704335-c37c-4db8-8475-bb21c037703a.png)
I just used the polyMesh from tutorial Onera M6 Wing and changed nothing. "Evaluating Objective Functions 000" and "Evaluating Objective Functions 001" didn't show such errors.
Did these errors lead to the above results? And how can they be fixed?

**Comment by friedenhe** [comment]
This means in the 2nd iteration, the design change is too large and the mesh quality is bad. You can try to reduce the upper and lower bounds of the design variables. Or use the mesh quality constraint in the tutorials/Ubend case.

**Comment by ayt1998** [comment]
Thank you very much, it was solved eventually.
